### ISSUE TITLE
DISCUSSION: x3d-tidy: No .x3d  or .x3dj output with this model run, trying to get used to filenaming conventions, JSON Schema discussion.

### ISSUE THREAD

**Issue by coderextreme** [create3000/x3d-tidy#17]
https://github.com/KhronosGroup/glTF-Sample-Assets/blob/main/Models/GlamVelvetSofa/glTF/GlamVelvetSofa.gltf

```
$ npx x3d-tidy@2.0.5 -i `pwd`/GlamVelvetSofa.gltf -o `pwd`/GlamVelvetSofaX_ITE.x3dj
(node:29452) ExperimentalWarning: CommonJS module C:\Users\jcarl\AppData\Roaming\npm\node_modules\npm\node_modules\debug\src\node.js is loading ES Module C:\Users\jcarl\AppData\Roaming\npm\node_modules\npm\node_modules\supports-color\index.js using require().
Support for loading ES Module in require() is an experimental feature and might change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
(node:23196) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
Cannot read properties of undefined (reading 'getExecutionContext')

jcarl@JustHumans MINGW64 ~/glTF-Sample-Models/2.0/GlamVelvetSofa/glTF (main)
$ ls -ltr
total 4736
-rw-r--r-- 1 jcarl 197609   17488 Sep  3 08:15 GlamVelvetSofa.gltf
-rw-r--r-- 1 jcarl 197609  124952 Sep  3 08:15 GlamVelvetSofa.bin
-rw-r--r-- 1 jcarl 197609 2487826 Sep  3 08:15 GlamVelvetSofa_normal.png
-rw-r--r-- 1 jcarl 197609  530651 Sep  3 08:15 GlamVelvetSofa_occlusion.png
-rw-r--r-- 1 jcarl 197609 1038409 Nov 27 19:47 GlamVelvetSofa.x3dj
-rw-r--r-- 1 jcarl 197609  638007 Nov 29 07:18 GlamVelvetSofaCGE.x3d
```

This worked!
```
$ npx x3d-tidy@2.0.6 -i file:/Users/jcarl/glTF-Sample-Models/2.0/GlamVelvetSofa/glTF/GlamVelvetSofa.gltf -o /c/Users/jcarl/glTF-Sample-Models/2.0/GlamVelvetSofa/glTF/GlamVelvetSofaX_ITE.x3dj
(node:13924) ExperimentalWarning: CommonJS module C:\Users\jcarl\AppData\Roaming\npm\node_modules\npm\node_modules\debug\src\node.js is loading ES Module C:\Users\jcarl\AppData\Roaming\npm\node_modules\npm\node_modules\supports-color\index.js using require().
Support for loading ES Module in require() is an experimental feature and might change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
(node:28608) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
(Use `node --trace-deprecation ...` to show where the warning was created)
(node:28608) V8: C:\Users\jcarl\AppData\Local\npm-cache\_npx\ecf4572556a06496\node_modules\x_ite\dist\assets\components\Texturing3DComponent.js:16 Invalid asm.js: Invalid member of stdlib
(node:28608) V8: C:\Users\jcarl\AppData\Local\npm-cache\_npx\ecf4572556a06496\node_modules\x_ite\dist\assets\components\Texturing3DComponent.js:53 Invalid asm.js: Invalid member of stdlib
```

**Comment by coderextreme** [comment]
Even when the JSON file is created, I don't know whether the schema is correct or not.  Most recent JSON schema is attached.
[x3d-4.0-debug.json](https://github.com/user-attachments/files/17960011/x3d-4.0-debug.json)
[x3d-4.0-JSONSchema.json](https://github.com/user-attachments/files/17960012/x3d-4.0-JSONSchema.json)

I am working out issues with validating glTF to .x3dj conversion, so I am trying to fix the JSON schema.

**Comment by coderextreme** [comment]
Yes, I did try the ‘pwd’/… with the latest.  See changes to     command line to make it work.

**Comment by coderextreme** [comment]
I have some JSON with the example In the message.  I converted the gltf to XML to try X3dToJson.xslt, but your version is better.  Your version is correct.  I don’t really want to spew Khronos’ IP everywhere.   There will probably never be an X3D JSON standard.  Probably what I’ll try to do is generate X3DJSAIL for the X_ITE component.

**Comment by coderextreme** [comment]
I’ve narrowed questionable stuff down to a screenful for X_ITE, but I realize that some things are hidden by other things.

**Comment by coderextreme** [comment]
What I have is a way to add X_ITE’s X3DUOM.xml into X3DOM (the main one).  I have a few things to add to a SimpleType to add X_ITE component, and addition to component Statement name enumeration

**Comment by coderextreme** [comment]
Okay.  I believe I found the issue in x3d-tidy.  Apparently, a Transform is included in the HAnimHumanoid.skeleton field, with both XML and JSON.

Here's the command-lines I used:
```
$ npx x3d-tidy@2.0.6 -i file:/Users/jcarl/glTF-Sample-Models/2.0/BrainStem/glTF/BrainStem.gltf -o /c/Users/jcarl/glTF-Sample-Models/2.0/BrainStem/glTF/BrainStem.x3dj

$ npx x3d-tidy@2.0.6 -i file:/Users/jcarl/glTF-Sample-Models/2.0/BrainStem/glTF/BrainStem.gltf -o /c/Users/jcarl/glTF-Sample-Models/2.0/BrainStem/glTF/BrainStem.x3d
```

This appears to be the issue:  <Transform DEF='_1' containerField='skeleton'
          rotation='-0.999999460009593 0.000679614892643972 0.000679614974962006 4.7123882718067'>

Transform isn't valid for the skeleton field, quoting X3D4:

MFNode     [in,out] skeleton              []       [HAnimJoint, HAnimSite]

Here's what the XML looks like, again, incorrect by my reading of X3D4:

        { "HAnimHumanoid":
          {
            "@DEF": "Poser-scene-root",
            "@name": "Poser_scene_root",
            "@skeletalConfiguration": "GLTF",
            "@rotation": [ 1, 0, 0, 4.71238898038469 ],
            "-skeleton": [
              { "Transform":
                {

Also, VRML:

DEF Poser-scene-root HAnimHumanoid {
  name "Poser_scene_root"
  skeletalConfiguration "GLTF"
  rotation 1 0 0 4.71238898038469
  skeleton DEF _1 Transform {

CGE reports:

$ ~/Downloads/castle-model-viewer-5.2.0-win64-x86_64/castle-model-viewer/castle-model-converter.exe --validate BrainStem
.x3d
castle-model-converter: Warning: X3D: Node "Transform" is not allowed in the field "skeleton" of the node "HAnimHumanoid"
castle-model-converter: Warning: X3D: Unknown X3D field name (unhandled X3D XML attribute) "description" in node "TimeSensor"
Exception "Exception":
Validation failed (consult the warnings above), exiting with non-zero status

(I'm going to have to talk to Michalis about the description in TimeSensor).

John
